src/vulkan-renderer/renderer.cpp

```diff
@@ -1087,24 +1101,27 @@ VkResult VulkanRenderer::create_descriptor_writes() {
     VkResult result = descriptor_manager->add_write_descriptor_set(global_descriptor, descriptor_writes[0]);
     vulkan_error_check(result);
 
-    return VK_SUCCESS;
-}
+    // Link the texture to the descriptor set so the shader can access it.
+    descriptor_image_info.imageLayout = VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL;
+    descriptor_image_info.imageView = textures[0].get_image_view();
```

Comment:
The vectors of textures, mesh_buffers, etc. may be empty, it is not guaranteed to have at least one value (or more).

Reply:
I will add `assert(!textures.empty());` to the beginning of `VulkanRenderer::create_descriptor_writes()`.

The problem here is that vulkan-renderer is not fully RAII-ified yet. This means we can't add a texture to the engine which will be constructed in the renderer's constructor (yet).

Therefore, we use `textures.emplace_back` to construct the texture in place.

Reply:
This will only partially solve it, asserts are not used in release builds.

Reply:
This issue will be resolved once we have fully RAII-ified the engine.